### PR TITLE
tso, server: refine the TSO allocator manager parameters

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #5895.

### What is changed and how does it work?

```commit-message
- Refine the TSO allocator manager parameters.
- Always run `tsoAllocatorLoop` to advance the Global TSO.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/tso/allocator_manager.go [hunk 1]
```diff
@@ -181,7 +191,7 @@ type AllocatorManager struct {
 	// in etcd, otherwise etcd will expire the leader key and other servers can campaign
 	// the primary/leader again. Etcd only supports seconds TTL, so here is second too.
 	leaderLease    int64
-	maxResetTSGap  func() time.Duration
+	maxResetTSGap  time.Duration
```
[the remark this answers is not in the document]
Comment (binshi-bing) [reply pkg/tso/allocator_manager.go:194]: If we change it, PD will have regression. In the future, TSO might need to support this hot configuration too.

**Review by binshi-bing — CHANGES_REQUESTED** [review]
Left critical comment -- if my understanding is correct.

File: pkg/tso/allocator_manager.go [hunk 2]
```diff
@@ -143,6 +143,16 @@ type ElectionMember interface {
 	PrecheckLeader() error
 }
 
+// ConfigProvider is used to provide TSO configuration.
+type ConfigProvider interface {
```
Comment (binshi-bing) [inline pkg/tso/allocator_manager.go:147]: It seems that there is no need to define a new interface. The newly added interface tso.ServiceConfig in pkg/tso/config.go was designed for this purpose. It's already used by tso.KeyspaceGroupManager and planned to use it for simplifying the interface of NewAllocatorManager too.

pd server config.Config just needs to implement  tso.ServiceConfig, i.e., only needs to add the following methods (the other methods have already been implemented, and GeBackendEndpoints() can be removed).

// GetName returns the Name
func (c *Config) GetName() string {
	return c.Name
}

// GetListenAddr returns the ListenAddr
func (c *Config) GetListenAddr() string {
	return c.ClientUrls
}

// GetAdvertiseListenAddr returns the AdvertiseListenAddr
func (c *Config) GetAdvertiseListenAddr() string {
	return c.AdvertiseClientUrls
}


// GetMaxResetTSGap returns the MaxResetTSGap.
func (c *Config) GetMaxResetTSGap() time.Duration {
	return ...
}

Comment (binshi-bing) [reply pkg/tso/allocator_manager.go:147]: I just noticed that, for PD/API GetMaxResetTSGap() needs to return s.persistOptions.GetMaxResetTSGap(). Yes, you're right to let Server implement the interface. Shall we combine ConfigProvider and tso.ServiceConfig into one interface? I'm ok to either way.

**Review by rleungx — APPROVED** [review]

File: pkg/tso/allocator_manager.go (hunk 2 above)
Comment (JmPotato) [reply pkg/tso/allocator_manager.go:147]: Sorry for merging this PR without noticing your comment. I filed a new PR to address this problem. PTAL.

https://github.com/tikv/pd/pull/6272

**Pull request merged**